src/app/public/modules/phone-field/phone-field.component.ts

```diff
@@ -0,0 +1,117 @@
+import {
+  ChangeDetectionStrategy,
+  Component,
+  EventEmitter
+} from '@angular/core';
+
+require('intl-tel-input/build/js/utils');
+require('intl-tel-input/build/js/intlTelInput');
+
+/**
+ * NOTE: We can not type these due the the current @types/intl-tel-input version having an
+ * undeclared type which causes linting errors.
+ */
+declare var intlTelInputUtils: any;
+declare var intlTelInputGlobals: any;
+
+import {
+  SkyCountryData
+} from './types';
+
+@Component({
+  selector: 'sky-phone-field',
+  templateUrl: './phone-field.component.html',
+  styleUrls: ['./phone-field.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class SkyPhoneFieldComponent {
+
+  public countryData: SkyCountryData[];
+
+  public defaultCountryData: SkyCountryData;
+
+  public disabled: boolean;
+
+  public set selectedCountry(newCountry: SkyCountryData) {
+    this._selectedCountry = newCountry;
+
+    if (!this._selectedCountry.placeholder) {
+      this._selectedCountry.placeholder = intlTelInputUtils.getExampleNumber(
+        newCountry.iso2,
+        true,
+        intlTelInputUtils.numberType.FIXED_LINE
+      );
+    }
+
+    this.countryData.splice(this.countryData.indexOf(newCountry), 1);
+
+    let sortedNewCountries = this.countryData
+      .sort((a, b) => {
+        if ((a === this.defaultCountryData || a.name < b.name) && b !== this.defaultCountryData) {
+          return -1;
+        } else {
+          return 1;
+        }
+      });
+
+    sortedNewCountries.splice(0, 0, newCountry);
+    this.countryData = sortedNewCountries;
+    this.selectedCountryChanged.emit(newCountry);
+  }
+
+  public get selectedCountry() {
+    return this._selectedCountry;
+  }
+
+  public selectedCountryChanged = new EventEmitter<SkyCountryData>();
```

Comment:
Need an `ngDestroy`? Also, if this is intended to be an emit that fires multiple times, it should be named in the present tense: `selectedCountryChange`.

Reply:
Not for the event emitter. That will destroy automatically. However, I did miss needing to add a `takeUntil` on the directive where it subscribes to that event. I've added that.

I also renamed the event.

Reply:
Actually, I went ahead and completed the event.